### ISSUE TITLE
single manager docker swarm stuck in Down state after reboot

### ISSUE THREAD

**Issue by alexanderkjeldaas** [moby/moby#34827]
**Description**

A single-manager swarm cluster gets into the following state after a abrupt reboot:

```
# docker node ls
ID                            HOSTNAME            STATUS              AVAILABILITY        MANAGER STATUS
ux3enftr6krhlrp8bbodo2q1l *   wsdocker6           Down                Active              Leader
```

The manager can't leave the swarm, as the `docker swarm leave --force` command times out.

The above state shouldn't be possible to be in.  By printing the above, the node must be a manager, and since there's only one manager, it has to infer that itself is that manager, and it thus can't be `Down`.

**Steps to reproduce the issue:**
1.  Create a swarm with 1 manager
2.  Abruptly reboot
3. Profit

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
 docker version
Client:
 Version:      17.06.2-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 20:00:17 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.2-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 19:59:11 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
root@wsdocker6:/etc/systemd/system# docker info
Containers: 83
 Running: 1
 Paused: 0
 Stopped: 82
Images: 103
Server Version: 17.06.2-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: pending
 NodeID: ux3enftr6krhlrp8bbodo2q1l
 Is Manager: true
 ClusterID: nhkjyh6xo2jdu2txkjt95xfua
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Root Rotation In Progress: false
 Node Address: x.x.x.x
 Manager Addresses:
  x.x.x.x:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 6e23458c129b551d5c9871e5174f6b1b7f6d1170
runc version: 810190ceaa507aa2727d7ae6f4790c76ec150bd2
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-93-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.63GiB
Name: wsdocker6
ID: KRY5:PQU2:2KPP:7QR6:4EXD:QFXT:KKVB:CZVN:OQUU:S5A3:Z5ZT:XQGN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: xxx
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by alexanderkjeldaas** [comment]
Some commands and their output:
```
# docker swarm init --force-new-cluster
Error response from daemon: context deadline exceeded
# docker node rm --force wsdocker6
Error response from daemon: This node is not a swarm manager. Worker nodes can't be used to view or modify cluster state. Please run this command on a manager node or promote the current node to a manager.
# docker swarm init
Error response from daemon: This node is already part of a swarm. Use "docker swarm leave" to leave this swarm and join another one.
# docker swarm leave
Error response from daemon: context deadline exceeded
# docker swarm leave --force
Error response from daemon: context deadline exceeded
```

**Comment by thaJeztah** [comment]
ping @aluzzardi @dperny PTAL

**Comment by alexanderkjeldaas** [comment]
Going to bed, when I woke up, the host was healthy.  Looking at the logs it seems like the host became healthy after 1 hour.
Again, guessing that either some time-out happened or some raft log processing that took a long time.

**Comment by abronan** [comment]
AFAIK manager membership is also tightly coupled to certificate management and rotation, the fact that the host could be added/removed normally after some time gives us a hint towards that direction. Also the time it took (*approximately 1 hour*), would point to the [expiration time](https://github.com/docker/swarmkit/blob/64aab747595cf08e7eff141acebaae1e3a96ccb5/ca/certificates.go#L72) to trigger a certificate rotation, thus why the node can be seen as healthy once the rotation is triggered: other managers were able to contact this node again.

Just speculation based on the information provided.

**Comment by Tkwon123** [comment]
Same issue applies to 17.09.0-ce. I clocked 2 hours (and 10 minutes) from my reboot to when the single master node registered as "ready"

**Comment by thaJeztah** [comment]
ping @nishanttotla PTAL as well if there's changes needed for this

**Comment by mahnunchik** [comment]
Have faced with the same issue on docker versions 17.12 and 18.03.

```
# docker node ls
ID                            HOSTNAME               STATUS              AVAILABILITY        MANAGER STATUS
fwlt7cd4d0lz9da14mbihs69s *   stage                  Down                Active              Leader
```

```
Swarm: pending
 NodeID: fwlt7cd4d0lz9da14mbihs69s
 Is Manager: true
 ClusterID: djxj99qu74s62tnzvfuknfy6z
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 10.0.8.20
 Manager Addresses:
  10.0.8.20:2377
```

**Comment by chris-fa** [comment]
Any updates on how to solve this?

**Comment by alexanderkjeldaas** [comment]
Got into this state again after last upgrade to docker 18.05

**Comment by alexanderkjeldaas** [comment]
Inspecting the node id I get among other things:
```
        },
        "Status": {
            "State": "down",
            "Message": "heartbeat failure",
            "Addr": "a.b.c.d"
        },
```

**Comment by Sialagio** [comment]
Hey everyone, digging out this thread.

Same issue as mentioned 24 days ago.
Anybody has a solution about it ?
I would like to avoid swarm leave/init operations which I did last time.

**Comment by Niedzwiedzw** [comment]
same as @alexanderkjeldaas , any solution?

**Comment by koubas** [comment]
Same here (sorry for spamming, but this thread needs to live!), in my case the nodes were down for about a day in our lab after a power outage and after powering it up today, all nodes (including one manager) are down with `"Message": "heartbeat failure"`.
Now, as I'm writing this, the manager node became Ready (after ~30min), the worker nodes are still down.

**Comment by abronan** [comment]
@koubas Are your workers still down? Try not to tinker with them and let us know if they do (or not) come back online after some time. Want to confirm my theory above (I may be completely off track here but without much information it's hard to say where the error comes from).

**Comment by koubas** [comment]
@abronan sorry, the manager problem was authentic, but nodes had different problem (rexray volume plugin I uninstalled recently appeared again in the logs and blocked swarm subsystem's start up), so I'd tinkered heavily already.
When it will be appropriate I'll break our lab intentionally again and I'll write the results, I'll gladly help with with resolving this.

**Comment by drnybble** [comment]
Seeing the same problem on 18.05-ce. I stopped the daemon for about 5 minutes using service docker stop. On service docker start I see:
ID                            HOSTNAME             STATUS              AVAILABILITY        MANAGER STATUS      ENGINE VERSION
nj4oo60bh6uvovtmehgucn34g     ions1   Ready               Active              Reachable           18.05.0-ce
k2quh59urx244fo487og0z669 *   ions2   Down                Active              Reachable           18.05.0-ce
s97wdji24ymbnuxs08snaedbq     ions3   Ready               Active              Leader              18.05.0-ce


        "Status": {
            "State": "down",
            "Message": "heartbeat failure for node in \"unknown\" state",
            "Addr": "0.0.0.0"
        },
 
I'll let it sit & see if it fixes itself.

**Comment by abronan** [comment]
@drnybble From the logs:

> `swarm component could not be started before timeout was reached`

There seems to be something preventing swarm to start on that Manager node you stopped. The logical followup is that when typing `docker nodes ls`, we then see that the node could not be reached because Swarm is not running (the `heartbeat failure` error).

It does not seem like this is the raft subsystem as it rightfully puts itself at the follower state and could process heartbeats (although it could be that it is restoring the state from a snapshot from the current leader in which case this could take some time and eat up on the time allocated for the startup).

The quorum isn't lost either as we have two Manager nodes at the `Ready` state.

By any chance, were you running heavy workloads on the remaining Manager nodes when this happened or were you issuing a lot of docker commands (which would build up the state quickly)?

In your case I would attempt a restart of the manager and see if it could finally start successfully.

/cc @cyli @dperny You may have a much better idea of what is going on here 😄

**Comment by drnybble** [comment]
Possible it was under load as tasks would have shifted to the running servers. I rebuilt my cluster using 18.06.0-ce and repeated the scenario of stopping and then starting the Docker service, but waiting for the load on the other nodes to settle down. Didn't encounter the problem!

**Comment by bartvanheukelom** [comment]
I experienced this same issue. Couldn't get it fixed, or at least not quickly enough, and decided to ditch Swarm for good old `docker-compose`, since I'm not actually using cluster features.

**Comment by mahnunchik** [comment]
Whats wrong with you? Single node swarm is unusable. It get stuck after each reboot.

**Comment by rbdiang** [comment]
Does anyone have any more information to offer on this issue?  We are seeing similar problem on a 5 node swarm cluster that has been working fine for months, we are on version 18.03.1-ce.  

Unfortunately the cluster is on an air gap network so it isn't easy to share /var/log/messages.  The leader status is 'Down' 3 other of the worker nodes are 'Ready'.  How can the manager be down, but some workers are up?

**Comment by Stephan-Walter** [comment]
I face the same problem from time to time, when I unplug the network cable for testing foutls. 

Most of the time, the master rejoins the swarm with Ready state and clean up all services, that has been started on another node. But sometimes, it stucks within Down state and let all the services running...

This is really a big problem, since this lead to the situation, that we have containers with the same IP running on different nodes, what breaks the setup...

**Comment by mhemrg** [comment]
Same issue here.  Docker version 17.12.1-ce, build 7390fc6

**Comment by mhemrg** [comment]
`journalctl` reports this line periodically:
```
dockerd[7260]: time="2019-01-18T17:18:44.208174709+01:00" level=error msg="fatal task error" error="task: non-zero exit (255)" module=node/agent/worker/taskmanager node.id=7mt7c9nxemwbx4e8ld6nyuxp0 service.id=ywd9vo8u7tjohtb1s3oohse8x task.id=r3exwedxyb93h5ay8wkdonim9
```

**Comment by ryandaniels** [comment]
For me, I had this problem with a swarm worker (on CentOS 7 - Docker version: 18.06.0-ce). 
The only error I could find was:

> swarm component could not be started before timeout was reached

Looking around in /var/lib/docker there was 1 huge file (besides the normal volumes etc).
tasks.db was 15GB!
```
[root@swarmwork-02 worker]# ll /var/lib/docker/swarm/worker/tasks.db 
-rw-r--r--. 1 root root 15726108672 Jan 25 13:32 /var/lib/docker/swarm/worker/tasks.db
```
I stopped docker, deleted the file and everything was fine.
Hopefully this helps others that come across this issue.

**Comment by dantin-s** [comment]
> For me, I had this problem with a swarm worker (on CentOS 7 - Docker version: 18.06.0-ce).
> The only error I could find was:
> 
> > swarm component could not be started before timeout was reached
> 
> Looking around in /var/lib/docker there was 1 huge file (besides the normal volumes etc).
> tasks.db was 15GB!
> 
> ```
> [root@swarmwork-02 worker]# ll /var/lib/docker/swarm/worker/tasks.db 
> -rw-r--r--. 1 root root 15726108672 Jan 25 13:32 /var/lib/docker/swarm/worker/tasks.db
> ```
> I stopped docker, deleted the file and everything was fine. The manager sees the worker again.
> Hopefully this helps others that come across this issue.

It seems work for me!

**Comment by djmaze** [comment]
Seeing a 1,5 GB `tasks.db` in my cluster which leads to swarm managers being stuck in the `Down' state after restart.

As Docker seems not to do this itself, is there a way to manually clean up old tasks from the db? I'd rather not delete the `tasks.db` altogether and risk getting [ghost tasks as described in the according swarmkit issue](https://github.com/docker/swarmkit/issues/2367#issuecomment-343825957).

**Comment by yorjaggy** [comment]
> For me, I had this problem with a swarm worker (on CentOS 7 - Docker version: 18.06.0-ce).
> The only error I could find was:
> 
> > swarm component could not be started before timeout was reached
> 
> Looking around in /var/lib/docker there was 1 huge file (besides the normal volumes etc).
> tasks.db was 15GB!
> 
> ```
> [root@swarmwork-02 worker]# ll /var/lib/docker/swarm/worker/tasks.db 
> -rw-r--r--. 1 root root 15726108672 Jan 25 13:32 /var/lib/docker/swarm/worker/tasks.db
> ```
> 
> I stopped docker, deleted the file and everything was fine. The manager sees the worker again.
> Hopefully this helps others that come across this issue.

It also works for me, I was a little bit skeptical but it works like a charm!

**Comment by misterunknown** [comment]
> For me, I had this problem with a swarm worker (on CentOS 7 - Docker version: 18.06.0-ce).
> The only error I could find was:
> 
> > swarm component could not be started before timeout was reached
> 
> Looking around in /var/lib/docker there was 1 huge file (besides the normal volumes etc).
> tasks.db was 15GB!
> 
> ```
> [root@swarmwork-02 worker]# ll /var/lib/docker/swarm/worker/tasks.db 
> -rw-r--r--. 1 root root 15726108672 Jan 25 13:32 /var/lib/docker/swarm/worker/tasks.db
> ```
> 
> I stopped docker, deleted the file and everything was fine. The manager sees the worker again.
> Hopefully this helps others that come across this issue.

This also worked for me, by I'm honestly very uncomfortable with that "solution". I guess the tasks.db file should get cleaned by the service itself. In my case the file was 1,3G. This should definititely get fixed.

**Comment by yuklia** [comment]
docker -v
Docker version 19.03.5, build 633a0ea838
one node swarm the same issue

**Comment by AeroSteveO** [comment]
> 

> 
> 
> > For me, I had this problem with a swarm worker (on CentOS 7 - Docker version: 18.06.0-ce).
> > The only error I could find was:
> > > swarm component could not be started before timeout was reached
> > 
> > 
> > Looking around in /var/lib/docker there was 1 huge file (besides the normal volumes etc).
> > tasks.db was 15GB!
> > ```
> > [root@swarmwork-02 worker]# ll /var/lib/docker/swarm/worker/tasks.db 
> > -rw-r--r--. 1 root root 15726108672 Jan 25 13:32 /var/lib/docker/swarm/worker/tasks.db
> > ```
> > 
> > 
> > I stopped docker, deleted the file and everything was fine. The manager sees the worker again.
> > Hopefully this helps others that come across this issue.
> 
> This also worked for me, by I'm honestly very uncomfortable with that "solution". I guess the tasks.db file should get cleaned by the service itself. In my case the file was 1,3G. This should definititely get fixed.
> 
> Edit: I run a cluster of 4 nodes, with Docker Engine 19.03.1

I run a cluster of 2 RPI 3B+ nodes and when my manager node stopped responding i forced a reboot. After the reboot it no longer recognized itself as being a manager node. I moved the tasks.db to a separate location in case i needed to move it back, but a service docker stop, mv tasks.db, service docker start fixed it for me.

TLDR: stopping docker, removing tasks.db, starting docker worked for me

**Comment by biliboo** [comment]
I'm also posting what worked from me.

_On each DOWN nodes_ 
docker swarm leave --force

_On the MOM_
docker node rm {id} (each nodes that are donw)
docker swarm join-token manager
docker swarm join-token worker

_On each nodes that where down_
Join the swarm with the command above depending if it's a manager or a worker.

_On the MOM_
docker node ls

I wish your nodes to be Ready again, or good luck otherwise.

**Comment by ggruber4711** [comment]
Solution to remove task.db worked for me aswell 👍

**Comment by soundstripe** [comment]
Removing tasks.db (on the node that will not rejoin) immediately resolved the problem for me as well.

**Comment by insantr** [comment]
> Seeing a 1,5 GB `tasks.db` in my cluster which leads to swarm managers being stuck in the `Down` state after restart (Docker 18.06.1-ce).
> 
> As Docker seems not to do this itself, is there a way to manually clean up old tasks from the db? I'd rather not delete the `tasks.db` altogether and risk getting [ghost tasks as described in the according swarmkit issue](https://github.com/docker/swarmkit/issues/2367#issuecomment-343825957).

This worked for me. Thanks @ryandaniels , you're awesome.
`Docker version 19.03.5, build 633a0ea`

**Comment by cpuguy83** [comment]
ping @dperny

**Comment by josepato** [comment]
> For me, I had this problem with a swarm worker (on CentOS 7 - Docker version: 18.06.0-ce).
> The only error I could find was:
> 
> > swarm component could not be started before timeout was reached
> 
> Looking around in /var/lib/docker there was 1 huge file (besides the normal volumes etc).
> tasks.db was 15GB!
> 
> ```
> [root@swarmwork-02 worker]# ll /var/lib/docker/swarm/worker/tasks.db 
> -rw-r--r--. 1 root root 15726108672 Jan 25 13:32 /var/lib/docker/swarm/worker/tasks.db
> ```
> 
> I stopped docker, deleted the file and everything was fine. The manager sees the worker again.
> Hopefully this helps others that come across this issue.

worked for me

**Comment by dpiquet** [comment]
> For me, I had this problem with a swarm worker (on CentOS 7 - Docker version: 18.06.0-ce).
> The only error I could find was:
> 
> > swarm component could not be started before timeout was reached
> 
> Looking around in /var/lib/docker there was 1 huge file (besides the normal volumes etc).
> tasks.db was 15GB!
> 
> ```
> [root@swarmwork-02 worker]# ll /var/lib/docker/swarm/worker/tasks.db 
> -rw-r--r--. 1 root root 15726108672 Jan 25 13:32 /var/lib/docker/swarm/worker/tasks.db
> ```
> 
> I stopped docker, deleted the file and everything was fine. The manager sees the worker again.
> Hopefully this helps others that come across this issue.

this also solved my single node swarm issue. Thank you. This should probably cleaned up internally in docker ?

**Comment by ertemmstein** [comment]
Many thanks, task.db was 59 gb! on my docker env.
Delete the file and the node is up again

**Comment by cpuguy83** [comment]
Not the first time I've seen people complain about a massive tasks.db

@dperny Do you know why this db would get so large? Is it possible we are leaking some data? Can it be pruned or compacted?

**Comment by thaJeztah** [comment]
There's a WIP PR in SwarmKit; https://github.com/docker/swarmkit/pull/2917

**Comment by warmanton** [comment]
Today we faced with exactly the same issue with swarm node in down state and big task.db file. We did upgrade docker from 19.03.5 to 19.03.8 on Ubuntu 18.04.4 LTS.  We have relatively small swarm installation but I noticed that task.db has different size from 8Mb to 32Mb on running nodes and it was 3Gb+ on down node. I resolved the issue by stopping docker , removing task.db and starting docker service back. Many thanks for that.

From my experience when any .db file has huge size then something will go wrong soon or already is broken.

**Comment by cpuguy83** [comment]
We've made a patch to help keep the task.db size down. It does not fix issues with already large task.db files. This should be in the next release.

**Comment by ryandaniels** [comment]
So far tasks.db looks good! No growth since upgrading to 19.03.9
https://docs.docker.com/engine/release-notes/#19039

**Comment by cpuguy83** [comment]
Closing since this should be resolved in 19.03.9

Thanks for reporting back!

**Comment by renepardon** [comment]
I've `Docker version 19.03.12, build 48a66213fe` and ran into the same problem with a single manager Swarm node. Deleting the **tasks.db** file worked for me as well. How comes that this file is so huge and keeps increasing and stops the node from being reachable?

**Comment by NealBozeman** [comment]
I can't believe that worked. My tasks.db was only 8.0mb. Restored all services on my node without side effect.

**Comment by cpuguy83** [comment]
@renepardon The file should not grow unbounded anymore, except for actual tasks running on the node. Old files were not cleaned up, though.

**Comment by 15981a0590** [comment]
> Solution to remove task.db worked for me aswell 👍

should in remove task.db file in worker node or leasder,as my leader is working fine and the status of my worker was down

**Comment by BreiteSeite** [comment]
I don't use moby but i found this issue after also having issues bringing up my single-node docker swarm after reboots. For me, deleting `/var/lib/docker/swarm/worker/tasks.db` worked as a workaround in the sense that i could successfully run `docker stack deploy` again (which otherwise didn't work as the swarm wasn't running), it not only felt wrong and dangerous but after automating that to be done before every docker-daemon start - i had another problem: the swarm didn't automatically start again after reboots (i guess this is stored inside the task.db somehow).

So i wanted to document my journey and learnings here. If this is the wrong place or there is a better place to document this, please let me know.

I'm using a raspberrypi 4b with Fedora 35 IoT Edition (aarch64)

I found that the problem is that the raspberry is extremely bad at keeping the time. My hardware clock was over 1 month behind. As i'm running Fedora - i don't know how or if i can sync the hardware clock from the Raspberry via Fedora (i assume that only works in raspbian?).

Anyway, i found out about the timedrift by checking journalctl:

```
Nov 15 00:00:31 rpi.fritz.box dockerd[815]: time="2021-11-15T00:00:31.869376273Z" level=error msg="cluster exited with error: error while validating Root CA Certificate: x509: certificate has expired or is not yet valid: current time 2021-11-15T00:00:31Z is before 2021-12-24T12:18:00Z"
Nov 15 00:00:31 rpi.fritz.box dockerd[815]: time="2021-11-15T00:00:31.869607826Z" level=error msg="swarm component could not be started" error="error while validating Root CA Certificate: x509: certificate has expired or is not yet valid: current time 2021-11-15T00:00:31Z is before 2021-12-24T12:18:00Z"
```

Notice how the logs are for 15th Nov (today is 24th Dec). I created the swarm today.
So the issue is that the certificates that docker creates internally for management of the swarm nodes are invalid as when the raspberrypi boots, it's thinking it's 15th Nov and the cert *begins* to be valid on the 24th Dec. So it's not yet valid.

Then i checked `journalctl -b` to see that a lot of log lines are from the past, even though the (re)boot happened just 5 minutes ago.

chronyd seems to be the daemon that Fedora uses by default to ensure correct system time. I disabled that and use the systemd native one: `systemctl stop chronyd && systemctl disable chronyd && systemctl enable systemd-timesyncd && systemctl start systemd-timesyncd`

Now i ensure the docker service waits for the system time to be set by adding time-sync.target to the `After=` section of the docker unit-file

```
[Unit]
After=network-online.target firewalld.service containerd.service time-sync.target
```

via `systemctl edit docker`.

I rebooted. It showed all docker log-entries with the correct time. However, my swarm still didn't persist between reboots. It showed:

```
Dec 24 13:41:01 rpi.fritz.box dockerd[820]: time="2021-12-24T13:41:01.720958879Z" level=error msg="error creating cluster object" error="name conflicts with an existing object" module=node node.id=uj13vzai54eeuyne7t2v1obcc```

Which i solved by completely tearing down my (single-node) swarm and recreating it:
`docker swarm leave --force` followed by a `docker swarm init [...]` command.

To my surprise my data in my containers was still kept - but please be sure what you are doing with your swarm config and don't blame me for data-loss.

Now with docker enabled (`systemctl enable docker`) my swarm automatically reboots with my machine - as it should be.

**Comment by aychernov** [comment]
Today i get the same issue: on root domain (via dashbord) give me 502 error, but my apps working fine.
I`m stopped and prune all container, pulling CapRover again and its help to me.
